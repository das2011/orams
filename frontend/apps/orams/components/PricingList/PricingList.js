import React from 'react'
import Alert from '@gov.au/page-alerts/lib/js/react.js'
import { uniqueID } from 'shared/utils/helpers'
import styles from './PricingList.scss'

const PricingList = props => {
  const { pricesData, successMessage, hideNav, serviceToEdit, supplier, isCeilingPriceUpdate } = props
  const { prices } = pricesData

  const renderSuccessMessage = () => {
    const message = isCeilingPriceUpdate ? 'Ceiling price updated' : 'Pricing updated'

    if (successMessage) {
      return (
        <Alert as="success">
          <h4>
            {message}
          </h4>
        </Alert>
      )
    }

    return ''
  }

  const renderHeader = () => {
    if (isCeilingPriceUpdate) {
      const supplierName = supplier ? supplier.name : ''

      return (
        <header>
          <h1 className="au-display-xl">
            {supplierName}
          </h1>
          <h2 className="au-display-lg">Update ceiling prices - select a region</h2>
          <div className={styles.stepTitle}>Step 2 of 3</div>
          <div
            role="button"
            tabIndex="0"
            className={styles.backLink}
            onClick={() => {
              hideNav(false)
              props.goToStep(1)
            }}
          >
            Back to supplier search
          </div>
        </header>
      )
    }

    return (
      <header>
        <h1 className="au-display-xl" tabIndex="-1">
          Pricing for {serviceToEdit.serviceName}
          <span>{serviceToEdit.subCategoryName ? ` (${serviceToEdit.subCategoryName})` : ''}</span>
        </h1>
        <div className={styles.stepTitle}>Step 2 of 4</div>
        <div
          role="button"
          tabIndex="0"
          className={styles.backLink}
          onClick={() => {
            hideNav(false)
            props.goToStep(1)
          }}
        >
          Back to services list
        </div>
      </header>
    )
  }

  return (
    <div className={styles.container}>
      {renderSuccessMessage()}
      {renderHeader()}
      <article role="main">
        <div className={styles.headingRow}>
          <div className="row">
            <div className="col-md-3 col-sm-3">Region</div>
            <div className="col-md-2 col-sm-2">Price cap</div>
            <div className="col-md-2 col-sm-2">
              Price <span className={styles.gstTitle}>inc GST</span>
            </div>
            <div className="col-md-2 col-sm-2">Start date</div>
            <div className="col-md-2 col-sm-2">End date</div>
            <div className="col-md-1 col-sm-1" />
          </div>
        </div>
        {prices &&
          prices.map((price, id = uniqueID()) =>
            <div key={id} className={styles.priceRow}>
              <div className="row">
                <div className="col-md-3 col-sm-3">
                  {`${price.region.state} ${price.region.name}`}
                </div>
                <div className="col-md-2 col-sm-2">
                  {`$${price.capPrice}`}
                </div>
                <div className="col-md-2 col-sm-2">
                  <span className={styles.price}>
                    {`$${price.price}`}
                  </span>
                </div>
                <div className="col-md-2 col-sm-2">
                  {price.startDate}
                </div>
                <div className="col-md-2 col-sm-2">
                  {price.endDate}
                </div>
                <div className="col-md-1 col-sm-1">
                  <div
                    role="button"
                    tabIndex="0"
                    className={styles.link}
                    onClick={() => {
                      props.editPrice(price)
                      window.scrollTo(0, 0)
                    }}
                  >
                    <strong>Edit</strong>
                  </div>
                </div>
              </div>
            </div>
          )}
      </article>
    </div>
  )
}

export default PricingList
