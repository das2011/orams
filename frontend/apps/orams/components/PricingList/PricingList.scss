.container {
  margin-bottom: 100px;
}

.stepTitle {
  font-size: 20px;
  color: #777777;
}

.backLink {
  font-size: 17px;
  font-weight: bold;
  margin-top: 15px;
  margin-bottom: 49px;
  cursor: pointer;
  text-decoration: underline;
}

.link {
  font-size: 17px;
  font-weight: bold;
  color: #202751;
  text-decoration: underline;
  cursor: pointer;
}

.headingRow {
  font-size: 17px;
  font-weight: bold;
  color: #202751;
  border-bottom: solid 3px #202751;
  padding-bottom: 7px;
}

.gstTitle {
  font-size: 14px;
  font-weight: normal;
}

.priceRow {
  padding-top: 13px;
  padding-bottom: 14px;
  border-bottom: solid 1px #777777;
  color: #313131;
}

.price {
  font-weight: bold;
  color: #0f979b;
}
