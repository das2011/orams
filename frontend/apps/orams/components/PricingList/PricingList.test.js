import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import expect from 'expect'
import snapshotRenderer from 'react-test-renderer'
import PricingList from './PricingList'

Enzyme.configure({ adapter: new Adapter() })

describe('#PricingList', () => {
  const gotoStepSpy = jest.fn()
  const defaultProps = {
    supplier: {
      name: 'Supplier Name'
    },
    serviceToEdit: {
      serviceName: 'Service To Edit',
      subCategoryName: 'Sub Category'
    },
    pricesData: {
      prices: [
        {
          capPrice: '1000',
          capPriceId: 1,
          price: '500',
          region: {
            name: 'Metro',
            state: 'NSW'
          },
          startDate: '01/01/2018',
          endDate: '31/12/3999'
        },
        {
          capPrice: '2000',
          capPriceId: 2,
          price: '1000',
          region: {
            name: 'Metro',
            state: 'NSW'
          },
          startDate: '01/01/2018',
          endDate: '31/12/3999'
        }
      ]
    },
    hideNav: jest.fn(),
    goToStep: gotoStepSpy,
    editPrice: jest.fn()
  }

  test('renders correct header when it is update ceiling price', () => {
    const isCeilingPriceUpdate = true
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate
    }
    const tree = snapshotRenderer.create(<PricingList {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correct header when it is not update ceiling price', () => {
    const isCeilingPriceUpdate = false
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate
    }
    const tree = snapshotRenderer.create(<PricingList {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders success message', () => {
    const isCeilingPriceUpdate = true
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate,
      successMessage: 'Success'
    }
    const tree = snapshotRenderer.create(<PricingList {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('goToStep() is called with correct parameters', () => {
    const isCeilingPriceUpdate = true
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate
    }
    const wrapper = mount(<PricingList {...props} />)

    wrapper.find('.backLink').simulate('click')

    expect(gotoStepSpy).toBeCalledWith(1)
  })

  test('editPrice() is called with correct parameters', () => {
    const editPriceSpy = jest.fn()
    const isCeilingPriceUpdate = true
    const props = {
      ...defaultProps,
      isCeilingPriceUpdate,
      editPrice: editPriceSpy
    }
    const wrapper = mount(<PricingList {...props} />)

    const priceRow = wrapper.find('.priceRow .link').first()
    priceRow.simulate('click')

    expect(editPriceSpy).toBeCalledWith(defaultProps.pricesData.prices[0])
  })
})
