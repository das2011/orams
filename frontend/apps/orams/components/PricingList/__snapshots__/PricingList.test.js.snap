// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#PricingList renders correct header when it is not update ceiling price 1`] = `
<div
  className="container"
>
  
  <header>
    <h1
      className="au-display-xl"
      tabIndex="-1"
    >
      Pricing for 
      Service To Edit
      <span>
         (Sub Category)
      </span>
    </h1>
    <div
      className="stepTitle"
    >
      Step 2 of 4
    </div>
    <div
      className="backLink"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Back to services list
    </div>
  </header>
  <article
    role="main"
  >
    <div
      className="headingRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          Region
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Price cap
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Price 
          <span
            className="gstTitle"
          >
            inc GST
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Start date
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          End date
        </div>
        <div
          className="col-md-1 col-sm-1"
        />
      </div>
    </div>
    <div
      className="priceRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          NSW Metro
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          $1000
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          <span
            className="price"
          >
            $500
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          01/01/2018
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          31/12/3999
        </div>
        <div
          className="col-md-1 col-sm-1"
        >
          <div
            className="link"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <strong>
              Edit
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      className="priceRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          NSW Metro
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          $2000
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          <span
            className="price"
          >
            $1000
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          01/01/2018
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          31/12/3999
        </div>
        <div
          className="col-md-1 col-sm-1"
        >
          <div
            className="link"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <strong>
              Edit
            </strong>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`#PricingList renders correct header when it is update ceiling price 1`] = `
<div
  className="container"
>
  
  <header>
    <h1
      className="au-display-xl"
    >
      Supplier Name
    </h1>
    <h2
      className="au-display-lg"
    >
      Update ceiling prices - select a region
    </h2>
    <div
      className="stepTitle"
    >
      Step 2 of 3
    </div>
    <div
      className="backLink"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Back to supplier search
    </div>
  </header>
  <article
    role="main"
  >
    <div
      className="headingRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          Region
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Price cap
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Price 
          <span
            className="gstTitle"
          >
            inc GST
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Start date
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          End date
        </div>
        <div
          className="col-md-1 col-sm-1"
        />
      </div>
    </div>
    <div
      className="priceRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          NSW Metro
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          $1000
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          <span
            className="price"
          >
            $500
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          01/01/2018
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          31/12/3999
        </div>
        <div
          className="col-md-1 col-sm-1"
        >
          <div
            className="link"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <strong>
              Edit
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      className="priceRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          NSW Metro
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          $2000
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          <span
            className="price"
          >
            $1000
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          01/01/2018
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          31/12/3999
        </div>
        <div
          className="col-md-1 col-sm-1"
        >
          <div
            className="link"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <strong>
              Edit
            </strong>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;

exports[`#PricingList renders success message 1`] = `
<div
  className="container"
>
  <div
    className="au-body au-page-alerts   au-page-alerts--success"
    role="alert"
  >
    <h4>
      Ceiling price updated
    </h4>
  </div>
  <header>
    <h1
      className="au-display-xl"
    >
      Supplier Name
    </h1>
    <h2
      className="au-display-lg"
    >
      Update ceiling prices - select a region
    </h2>
    <div
      className="stepTitle"
    >
      Step 2 of 3
    </div>
    <div
      className="backLink"
      onClick={[Function]}
      role="button"
      tabIndex="0"
    >
      Back to supplier search
    </div>
  </header>
  <article
    role="main"
  >
    <div
      className="headingRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          Region
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Price cap
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Price 
          <span
            className="gstTitle"
          >
            inc GST
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          Start date
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          End date
        </div>
        <div
          className="col-md-1 col-sm-1"
        />
      </div>
    </div>
    <div
      className="priceRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          NSW Metro
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          $1000
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          <span
            className="price"
          >
            $500
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          01/01/2018
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          31/12/3999
        </div>
        <div
          className="col-md-1 col-sm-1"
        >
          <div
            className="link"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <strong>
              Edit
            </strong>
          </div>
        </div>
      </div>
    </div>
    <div
      className="priceRow"
    >
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3"
        >
          NSW Metro
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          $2000
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          <span
            className="price"
          >
            $1000
          </span>
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          01/01/2018
        </div>
        <div
          className="col-md-2 col-sm-2"
        >
          31/12/3999
        </div>
        <div
          className="col-md-1 col-sm-1"
        >
          <div
            className="link"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <strong>
              Edit
            </strong>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;
